### PR TITLE
Redesign: Apply redesign

### LINKED ISSUE #3: Redesign groundkeeper UI
[NOTE]: # ( ^^ Provide a general summary of the request in the title above. ^^ )

## Summary
It doesn't look like this nerdpack, as useful as it is, has had much design attention up to this point. A thoughtful redesign would improve the user experience by quite it a bit, so I'm going to take some time to work on a redesign that I can recommend in a PR.

[NOTE]: # ( Provide a brief overview of what the new feature is all about. )

## Desired Behaviour
Exactly the behavior that it already has, but with the application of ui design best practices and a visual style that aligns matches that of NR1.

[NOTE]: # ( Tell us how the new feature should work. Be specific. )
[TIP]:  # ( Do NOT give us access or passwords to your New Relic account or API keys! )

## Possible Solution
A redesign! Wireframe and mockups to come in the comments...

[NOTE]: # ( Not required. Suggest how to implement the addition or change. )

## Additional context
This matters to me because I think it's important that the open source applications that we release as an org meet our standard of design and UX. Most/if not all of the folks building these apps aren't designers. This happens to be my specialty, so here we go!

[TIP]:  # ( Why does this feature matter to you? What unique circumstances do you have? )

### PR DESCRIPTION
**Pull request by danielgolden**
#### Redesign toolbar:
- [x] Adds styles for the new toolbar
- [x] Swaps out old `<select>` elements for `<Dropdown>` components
- [x] Refactors filter functions to work with new dropdowns
- [x] Adds new "table state" Dropdown into toolbar

#### Redesign tables:
- [x] Feed the new table component ([React Bootstrap Table 2](https://react-bootstrap-table.github.io/react-bootstrap-table2/)) data in the format that needs.
- [x] Add search to the top of the primary table
- [x] Replace the primary table with the new component
- [x] Replace the agent versions table with the new component
- [x] Style tables according to latest mockup from #3.

### Screenshot
Captura de Pantalla 2020-01-31 a la(s) 9 27 11 a  m

---

Resolves #3

### CODE REVIEW

File: nerdlets/groundskeeper-nerdlet/components/TableWrapper.js [hunk 1]
```diff
@@ -19,82 +18,15 @@ export default class TableWrapper extends React.PureComponent {
...
+    const { tableData } = this.props;
 
     return (
-      <table>
-        <thead>
-          <tr>{headers}</tr>
-        </thead>
-        <tbody>{rows}</tbody>
-      </table>
+      <BootstrapTable
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/components/TableWrapper.js]: This is so simple now, seems like you could remove the tablewrapper component entirely.

File: nerdlets/groundskeeper-nerdlet/nrds.scss [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    color: #8e9494;
+  }
+  :-moz-placeholder {
+    /* Firefox 18- */
+    color: #8e9494;
+  }
+  
+  // ==============================================================
+  // Checkboxes
+  // ==============================================================
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/nrds.scss]: Why the custom styles on checkboxes? Can we just inherit the styles provided by the NR1 SDK?

File: nerdlets/groundskeeper-nerdlet/nrds.scss [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+          background-color: #f4f5f5;
+          border-color: #d5d7d7;
+        }
+      }
+    }
+  }
+  
+  // ==============================================================
+  // Radio Buttons
+  // ==============================================================
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/nrds.scss]: Same question as for checkbox - can we just use the standard NR1 styles?

File: nerdlets/groundskeeper-nerdlet/nrds.scss [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        background-color: #d5d7d7;
+        top: 4px;
+        left: 4px;
+      }
+    }
+  }
+  
+  // ==============================================================
+  // Typography
+  // ==============================================================
```
Comment (fightingmonk) [inline nerdlets/groundskeeper-nerdlet/nrds.scss]: Same question as for checkbox - can we just use the standard NR1 styles for all of our typography?

File: nerdlets/groundskeeper-nerdlet/components/TableWrapper.js (hunk 1 above)
Comment (danielgolden) [reply nerdlets/groundskeeper-nerdlet/components/TableWrapper.js]: Good point. Looks like it wasn't even being used after these changes. Pulling it out now.

**Comment by danielgolden** [comment]
We spoke about the rationale behind removing the NRDS.scss stylesheet in slack. This addresses 3 of the comments from the code review, which I have since resolved.

**Comment by fightingmonk** [comment]
Looks really good! Performance is much improved and the layout bugs in Firefox are gone. I'm going to merge this.

One followup request: sorting by age looks like it's an alphabetical search. Can you add a numeric sort field for that column so (e.g.) `99 weeks` sorts lower than `100 weeks`? For unknown `?` entries, maybe use -1 as the sort value?

**Pull request merged**